pkg/gui/controllers/helpers/fixup_helper.go

```diff
@@ -0,0 +1,175 @@
+package helpers
+
+import (
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/jesseduffield/generics/set"
+	"github.com/jesseduffield/lazygit/pkg/commands/models"
+	"github.com/jesseduffield/lazygit/pkg/gui/types"
+	"github.com/jesseduffield/lazygit/pkg/utils"
+	"github.com/samber/lo"
+)
+
+type FixupHelper struct {
+	c *HelperCommon
+}
+
+func NewFixupHelper(
+	c *HelperCommon,
+) *FixupHelper {
+	return &FixupHelper{
+		c: c,
+	}
+}
+
+type deletedLineInfo struct {
+	filename     string
+	startLineIdx int
+	numLines     int
+}
+
+func (self *FixupHelper) HandleFindBaseCommitForFixupPress() error {
+	diff, useIndex, err := self.getDiff()
+	if err != nil {
+		return err
+	}
+	if diff == "" {
+		return self.c.ErrorMsg("No changes to commit")
+	}
+
+	deletedLineInfos := self.parseDiff(diff)
+	if len(deletedLineInfos) == 0 {
+		return self.c.ErrorMsg("No deleted lines in diff")
+	}
+
+	shas := self.blameDeletedLines(deletedLineInfos)
+
+	if len(shas) == 0 {
+		// This should never happen
+		return self.c.ErrorMsg("No base commits found")
+	}
+	if len(shas) > 1 {
+		subjects, err := self.c.Git().Commit.GetShasAndCommitMessagesFirstLine(shas)
+		if err != nil {
+			return err
+		}
+		return self.c.ErrorMsg("Multiple base commits found. (" +
+			lo.Ternary(useIndex, "Try staging fewer changes", "Try staging some of the changes") +
```

Comment:
Saying 'Try staging some of the changes' isn't very clear: it may not be obvious that this action only looks at staged changes if they're present. Perhaps: 'If you stage some of the changes, we'll only consider those staged changes when looking for the base commit, so we're more likely to find a single match". Perhaps it's even worth spelling that out in the case where there are already staged changes.

Reply:
I'm not convinced; this additional information might be useful the very first time you run the command and are not yet familiar with how it works. After that, the important information is the list of subjects below, and the longer text above makes this stand out a lot less and makes it harder to grok the information. (I just tried it with the longer message, and didn't like it.)